From a19b53082f27c8af402d5284694beb7b6bc8bdfc Mon Sep 17 00:00:00 2001
From: Amin Bandali <bandali@gnu.org>
Date: Sun, 26 Feb 2023 01:00:03 -0500
Subject: [PATCH] Use dh_auto_* tools for the client

Also the stopgap sed workaround for building without Qt WebEngine is
no longer necessary.  The issue was calling out to dh_auto_configure
but without the right flags, which would then overwrite/interfere with
the flags specified in the earlier manual call to cmake.
---
 debian/rules | 36 ++++++++++++------------------------
 1 file changed, 12 insertions(+), 24 deletions(-)

diff --git a/debian/rules b/debian/rules
index f414d029..03522721 100755
--- a/debian/rules
+++ b/debian/rules
@@ -81,18 +81,9 @@ endif
 	    --disable-shared
 
 # Qt client configure
-# Stopgap workaround to disable building with Qt WebEngine
-	sed -i 's/set(WITH_WEBENGINE true)/#&/' CMakeLists.txt
-	mkdir build
-	cd build && \
-	  cmake \
-	    -DCMAKE_INSTALL_PREFIX=/usr \
-	    -DCMAKE_BUILD_TYPE=Release \
-	    -DWITH_DAEMON_SUBMODULE=true \
-	    -DWITH_WEBENGINE=false \
-	    ..
-
-	dh_auto_configure
+	dh_auto_configure -- \
+	  -DWITH_DAEMON_SUBMODULE=true \
+	  -DWITH_WEBENGINE=false
 
 override_dh_auto_build:
 # Daemon build
@@ -100,7 +91,7 @@ override_dh_auto_build:
 	pod2man daemon/man/jamid.pod > daemon/jamid.1
 
 # Qt client build
-	make -C build -j$(NO_CPUS) V=1
+	dh_auto_build
 
 override_dh_auto_clean:
 # Generated contrib tarballs
@@ -113,9 +104,7 @@ override_dh_auto_clean:
 	rm -rfv daemon/jamid.1
 
 # Qt client clean
-	[ -f build/Makefile ] && \
-	  make -C build clean || true
-	rm -rfv build
+	dh_auto_clean
 
 override_dh_auto_install:
 # Daemon install
@@ -129,8 +118,13 @@ override_dh_auto_install:
 	rm -rfv $(CURDIR)/debian/jami/usr/lib/cmake
 
 # Qt client install
-	cd build && \
-	  make DESTDIR=$(CURDIR)/debian/jami install
+	dh_auto_install --destdir=$(CURDIR)/debian/jami
+
+# Work around dak not liking files with timestamps too far in the past
+execute_before_dh_strip_nondeterminism:
+	find $(CURDIR)/debian/jami/ ! -newermt "jan 01, 2000" \
+	  -exec touch --date="@$(SOURCE_DATE_EPOCH)" {} +
+
 
 tmp_dir := $(shell mktemp -d)
 version := $(shell dpkg-parsechangelog -ldebian/changelog | perl -ne \
@@ -139,12 +133,6 @@ ds := $(shell dpkg-parsechangelog -ldebian/changelog | perl -ne \
               'print $$1 if m{^Version:\s+(?:\d+:)?(?:\d.*)(\~ds.+)(?:\-\d+.*)};')
 pj := $(tmp_dir)/jami-*/daemon/contrib/tarballs-unpacked/pjproject-*/pjproject-*
 
-# Work around dak not liking files with timestamps too far in the past
-execute_before_dh_strip_nondeterminism:
-	find $(CURDIR)/debian/jami/ ! -newermt "jan 01, 2000" \
-	  -exec touch --date="@$(SOURCE_DATE_EPOCH)" {} +
-
-
 # Repack the tarball with contrib tarballs unpacked
 get-orig-source:
 # Download jami tarball
-- 
GitLab

